src/Microsoft.AspNet.Mvc.TagHelpers/Internal/PartialModeMatchLoggerStructure.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    /// <summary>
+    /// An <see cref="ILoggerStructure"/> for log messages regarding <see cref="ITagHelper"/> instances that opt out of
+    /// processing due to missing attributes for one of several possible modes.
+    /// </summary>
+    public abstract class PartialModeMatchLoggerStructure : ILoggerStructure
+    {
+        /// <summary>
+        /// The log message.
+        /// </summary>
+        public string Message
+        {
+            get
+            {
+                return "Tag Helper has missing required attributes.";
+            }
+        }
+
+        /// <summary>
+        /// The values associated with the structured log message.
+        /// </summary>
+        protected IEnumerable<KeyValuePair<string, object>> Values { get; set; }
```

Comment:
Odd having a `{ get; set; }` for an `IEnumerable<KeyValuePair<string, object>`, can't this just be a private and take this value in via the ctor.